src/agentscope/service/inspector/file_inspector.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+"""
+The ServiceFunction utils for inspecting files.
+Convert different file formats to LLM understandable text.
+"""
+from io import StringIO
+import os
+from typing import Optional, Union, Dict, Callable
+
+import mammoth
+import puremagic
+import markdownify
+from bs4 import BeautifulSoup
+import pandas as pd
+import pdfminer
+import pdfminer.high_level
+from agentscope.service import ServiceResponse, ServiceExecStatus
+
+
+def _guess_ext_magic(file_path: str) -> Union[str, None]:
+    """
+    Use puremagic to guess a file's extension based on the first few bytes.
+    """
+    guesses = puremagic.magic_file(file_path)
+    if len(guesses) > 0:
+        ext = guesses[0].extension.strip()
+        if len(ext) > 0:
+            return ext
+
+    return None
+
+
+# currently supported file types
+# plain_text, docx, xlsx, pdf, html, py, txt
+# TODO support more types: image, audio, etc.
+def inspect_file_as_text(file_path: str) -> ServiceResponse:
+    """Inspect common-types files to markdown style text,
+    so the llm can inspect such files.
+    Currently support '.docx', '.xlsx', '.pdf', '.txt', '.py', '.html', etc.
+    Args:
+        file_path (str):
+            the path of the file to inspect
+    """
+    try:
+        if not os.path.isfile(file_path):
+            raise FileNotFoundError(f"No such file: '{file_path}'")
+
+        _, ext = os.path.splitext(file_path)
+        ext = ext.lower()
+
+        if not ext:
+            ext = _guess_ext_magic(file_path)
+            if ext:
+                ext = "." + ext.lower()
+
+        inspect_functions: Dict[str, Callable[[str], ServiceResponse]] = {
+            ".docx": inspect_docx_as_text,
+            ".xlsx": inspect_xlsx_as_text,
+            ".pdf": inspect_pdf_as_text,
+            ".html": inspect_html_as_text,
```

Comment:
Should we remove the existing html parsing function? The two functions are duplicated in functionality.